build.gradle.kts

```diff
@@ -0,0 +1,235 @@
+import com.adarshr.gradle.testlogger.theme.ThemeType
+import org.jetbrains.changelog.markdownToHTML
+import org.jetbrains.intellij.tasks.RunPluginVerifierTask.VerificationReportsFormats.*
+import org.jetbrains.intellij.tasks.RunPluginVerifierTask.FailureLevel.*
+
+fun properties(key: String) = providers.gradleProperty(key)
+fun environment(key: String) = providers.environmentVariable(key)
+fun prop(name: String): String {
+    return properties(name).get()
+}
+plugins {
+    java // Java support
+    alias(libs.plugins.kotlin) // Kotlin support
+    alias(libs.plugins.gradleIntelliJPlugin) // Gradle IntelliJ Plugin
+    alias(libs.plugins.changelog) // Gradle Changelog Plugin
+    alias(libs.plugins.testLogger) // Nice test logs
+    jacoco // Code coverage
+}
+
+group = prop("pluginGroup")
+version = prop("pluginVersion")
+
+val lsp4jVersion = prop("lsp4jVersion")
+
+// Configure project's dependencies
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven { url = uri("https://repository.jboss.org/nexus/content/repositories/snapshots") }
+    maven { url = uri("https://repository.jboss.org/nexus/content/groups/public") }
+    maven { url = uri("https://repo.eclipse.org/content/repositories/lsp4mp-snapshots") }
+    maven { url = uri("https://repo.eclipse.org/content/repositories/lsp4mp-releases") }
+    maven { url = uri("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
+}
+
+val lsp: Configuration by configurations.creating
+
+sourceSets {
+    named("test") {
+        java.srcDir("src/test/java")
+        java.srcDir("intellij-community/java/testFramework/src")
+        java.srcDir("intellij-community/platform/external-system-api/testFramework/src")
+        java.srcDir("intellij-community/platform/external-system-impl/testSrc")
+        java.srcDir("intellij-community/platform/lang-impl/testSources")
+        java.srcDir("intellij-community/platform/testFramework/extensions/src")
+        java.srcDir("intellij-community/plugins/gradle/src")
+        java.srcDir("intellij-community/plugins/gradle/testSources")
+        java.srcDir("intellij-community/plugins/gradle/tooling-extension-impl/testSources")
+        java.srcDir("intellij-community/plugins/maven/src/test/java")
+        java.srcDir("intellij-community/plugins/maven/testFramework/src")
```

Comment:
those should not be needed, as they're specific to testing based on import of Maven and Gradle projects

Reply:
removed